src/user/components/SignUpCard.vue

```diff
@@ -0,0 +1,198 @@
+<template>
+  <div class="card">
+    <div class="signup">
+      <heading><h3 class="signup--heading">Sign Up</h3></heading>
+
+      <form class="form" @submit.prevent="createAccount()">
+        <div class="form--element">
+          <label for="name" class="label">Name:</label>
+          <TextInput
+            id="name"
+            v-model="newAccount.name"
+            name="name"
+            type="text"
+            required
+            class="input"
+          />
+        </div>
+
+        <div class="form--element">
+          <label for="email" class="label">Email:</label>
+          <TextInput
+            id="email"
+            v-model="newAccount.email"
+            name="email"
+            type="email"
+            required
+            class="input"
+          />
+        </div>
+
+        <div class="form--element">
+          <label for="password" class="label">Password:</label>
+          <TextInput
+            id="password"
+            ref="pwRef"
+            v-model="newAccount.pw.password"
+            name="password"
+            type="password"
+            minlength="8"
+            required
+            class="input"
+            @change="validatePassword"
+          />
+        </div>
+
+        <div class="form--element">
+          <label for="passwordConfirm" class="label">Confirm Password:</label>
+          <TextInput
+            id="passwordConfirm"
+            v-model="newAccount.pw.confirm"
+            name="passwordConfirm"
+            type="password"
+            minlength="8"
+            required
+            class="input"
+            @change="validatePassword"
+          />
+        </div>
+
+        <AppButton variant="primary" type="submit" class="btn">
+          Create account
+        </AppButton>
+
+        <AppButton
+          variant="secondary"
+          type="button"
+          class="btn"
+          @click="logInViaGoogle()"
+        >
+          <GoogleLogo variant="primary" class="btn-logo" />
+          Login with Google
+        </AppButton>
+      </form>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, ref, computed } from "vue";
+import AppButton from "@/common/AppButton.vue";
+import TextInput from "@/common/TextInput.vue";
+import GoogleLogo from "@/common/app-icon/GoogleLogo.vue";
+import { useRouter, useRoute } from "vue-router";
+import client from "../client";
+
+export default defineComponent({
+  name: "SignUpCard",
+  components: { AppButton, TextInput, GoogleLogo },
+
+  setup() {
+    const router = useRouter();
+    const route = useRoute();
+
+    const newAccount = ref({
+      name: "",
+      email: "",
+      pw: {
+        password: "",
+        confirm: ""
+      }
+    });
+
+    // const pwRef = ref<InstanceType<typeof TextInput>>();
+    const pwRef = ref();
```

Comment:
Can you rename these to `password`? It's a bit more legible I think, took me a sec to figure out what `pw` was.

Reply:
Got it